### PR TITLE
Dozenal Time 2

### PR DESCRIPTION
**Pull request by numerist**
Not familiar enough with this process. Wanting the latest version(s) to be approved. Conflict has been resolved AFAIK.

### CODE REVIEW

**Comment by numerist** [comment]
Conflict has been resolved AFAIK.

**Comment by numerist** [comment]
Conflict has been resolved AFAIK.

File: apps.json [hunk 1]
```diff
@@ -4702,6 +4702,37 @@
...
+  "tags":"clock",
+  "type":"clock",
+  "allow_emulator":true,
+  "readme": "README.md",
+  "storage": [
+    {"name":"doztime.app.js","url":"app.js"},
+    {"name":"doztime.img","url":"app-icon.js","evaluate":true}
+  ]
+}
+]
```
Comment (myxor) [inline apps.json:4720]: There is a syntax error here

**Comment by numerist** [comment]
The item below was approved some months ago and has not changed. It is therefore a mystery that there should be a syntax error.

There may have been a problem with the app doztime2, which is new. I think that I have fixed that problem.

I am interested in having doztime2 reviewed again. I am not a coding expert; the code was written by someone else, who is no longer available. But with an indication of specifically what is wrong, I should be able to correct any problem.

Thank you,

—
Numerist
(Paul Rapoport)

**Comment by myxor** [comment]
There is a JSON syntax error in the apps.json file.
You can always check the JSON syntax by copying the content of the apps.json file on pages like www.jsonlint.com.
I suggest you to try it out and find the error.

**Comment by numerist** [comment]
Thank you. Some brackets were misplaced. According to www.jsonlint.com, the doztime and doztime2 apps.json files are now correct.

I ask, therefore, that the doztime2 submission be reviewed in its entirety now, for consideration for bangle.js 2.

—
Numerist

**Comment by myxor** [comment]
Plesse have a look at the job log from Travis at the bottom of this pull request.
You can see that the job failed.
The log of your last commit is here:
 
https://app.travis-ci.com/github/espruino/BangleApps/builds/243756877
It has a few warnings and an error:

```Warning: App gpsrec data file wildcard .gpsrc? does not include app ID

Warning: JS icon s7clk.img should be 48x48px (or slightly under) but is instead 50x50px

Warning: JS icon snek.img should be 48x48px (or slightly under) but is instead 64x64px

Warning: JS icon astral.img should be 48x48px (or slightly under) but is instead 50x50px

Warning: JS icon alpinenav.img should be 48x48px (or slightly under) but is instead 50x50px

Warning: JS icon slomoclock.img should be 48x48px (or slightly under) but is instead 64x64px

Warning: JS icon arrow.img should be 48x48px (or slightly under) but is instead 50x50px

Warning: App hourstrike uses storage file hourstrike.json instead of data file

ERROR: App doztime has no 'supports' field or it's not an array

The command "npm test" exited with 1.```

There you can find what needs to be changed.
* Your icons have non-optimal sizes
* You are missing the field 'supports' in the JSON definition of your app. Please have a look at the example app or other apps how the definition has to look

**Comment by numerist** [comment]
Thanks again. The icon size problem does not apply; my icon is 40x40. However, the “supports” field has been added to apps.json, and I request a new review of the app doztime2 (Dozenal Time 2).

—
Numerist

**Comment by numerist** [comment]
I created a new pull request to submit this app to the main (master) bangle.js app loader. I trust that's the correct procedure. Thanks again.

**Comment by numerist** [comment]
Jan. 1 2022: changed the positioning of two digits in the displays slightly and kept non-backlit display on for 24 hours

**Comment by gfwilliams** [comment]
Thanks! just wanted to check - the idea here is just to have a version of Dozenal time that works for Bangle.js 2?

Because I think there's a tidier way of handling it - having one app with two different source files, one for bangle.js 1 and one for 2. Assuming that's fine I'm happy to make the changes here.

**Comment by gfwilliams** [comment]
Just merging, but I'm not entirely sure what's going on here... Looks like the original dozenal time hasn't changed, but we now have 2 new files:

* apps/doztime2.app.js 
* apps/doztime2/app.js

They are both slightly different. Any idea which one should be used for what?

**Comment by gfwilliams** [comment]
Actually forget that - I'm just using `apps/doztime2/app.js`

**Comment by numerist** [comment]
Thanks. If I were a coder, etc., I’d understand what’s going on. But unfortunately I’m not.

The correct file will have this towards the end:

timeActiveUntil.setSeconds(timeActiveUntil.getSeconds()+86400);

It’s the 86400 that’s important.

**Comment by gfwilliams** [comment]
Ok, it was the other file then! Just put that one in

**Comment by numerist** [comment]
Looks good now. Thanks very much for your work, not to mention patience and understanding.

**Pull request merged**